aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -1168,6 +1178,8 @@ TORCH_IMPL_FUNC(scatter_src_out)
  const Tensor& index,
  const Tensor& src,
  const Tensor& out) {
+  dim = maybe_wrap_dim(dim, self.dim());
+  if (index.numel() == 0) return;
```

Comment:
Since every `scatter` IMPL function calls `scatter_impl` (which does all the computation), what do you think about moving this line there? Then, you can remove the other `scatter` meta function checks, too!

Reply:
The idea of having `index.numel() == 0` here was to reduce one dispatch when an empty index tensor is passed. But I think, from the RFC of structured kernels, reducing redundancy is the first task as long as it doesn't affect the performance, and this one dispatch doesn't cost us a lot. So, I agree with you, this has been moved to IMPL function now. :) Thanks!